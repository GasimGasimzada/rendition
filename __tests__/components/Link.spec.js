/* globals expect, describe, it */
import { mount } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'
import Provider from '../../src/components/Provider'
import Link from '../../src/components/Link'

describe('Link component', () => {
  it('Should match the stored snapshot', () => {
    const component = renderer.create(
      <Provider>
        <Link>Link</Link>
      </Provider>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Should display the href prop if a child is not provided', () => {
    const url = 'https://google.com'
    const component = mount(
      <Provider>
        <Link href={url} />
      </Provider>
    )

    expect(component.find(`a [href="${url}"]`).text()).toEqual(url)
  })
})
