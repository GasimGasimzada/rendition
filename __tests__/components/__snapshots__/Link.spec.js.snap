// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component Should match the stored snapshot 1`] = `
<div
  className="sc-bdVaJa jKopLr"
>
  <a
    className="sc-jTzLTM kVIkyB sc-fjdhpX jDBYsh"
    color="primary.main"
  >
    Link
  </a>
</div>
`;
